app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -338,17 +413,45 @@ fun DisplayParticipants(trip: Trip, userViewModel: UserViewModel) {
   }
 }
 
-// Helper function to convert Timestamp to String format.
+/** Helper function to convert Timestamp to String format. */
 fun Timestamp.toDateString(): String {
   val sdf = java.text.SimpleDateFormat("MMM dd yyyy", java.util.Locale.getDefault())
   return sdf.format(this.toDate())
 }
 
-// Helper function to generate the correct string
+/** Helper function to generate the correct string for the number of participants */
 fun generateParticipantString(numberOfParticipants: Int): String {
   return when (numberOfParticipants) {
     0 -> "No participants."
     1 -> "1 Participant:"
     else -> "$numberOfParticipants Participants:"
   }
 }
+/**
+ * Opens the default calendar app on the device with pre-filled details for a new event.
+ *
+ * This method creates an `Intent` to insert a calendar event using the trip's details such as name,
+ * description, start time, and end time.
+ *
+ * @param context The context used to launch the intent.
+ * @param trip The trip data used to populate the calendar event details.
+ */
+internal fun openGoogleCalendar(context: Context, trip: Trip) {
```

Comment:
Thank you for adding the documentation for this function! However, I was wondering what's the supposed behavior when no calendar app is found or when there's an error

Reply:
I've just solved this issue, it's in my latest commit. Now when there's an error/no calendar app is found, selecting the menu item just displays a toast saying there's no compatible calendar app.